core/vm/logger.go

```diff
@@ -65,36 +58,34 @@ type StructLog struct {
 	Err     error
 }
 
+type VmLogger interface {
+	CaptureState(env Environment, pc uint64, op OpCode, gas, cost *big.Int, memory *Memory, stack *stack, contract *Contract, depth int, err error);
+}
```

Comment:
I think i'd prefer `TraceInstruction` as the method name because more trace events might be added later.